src/transformers/visitors/toOrchestrationVisitor.ts

```diff
@@ -409,6 +409,59 @@ const visitor = {
           );
         }
         }
+      } else if (!scope.modifiesSecretState()){
+        const contractName = `${parent.name}Shield`;
+        const fnName = path.getUniqueFunctionName();
+        node.fileName = fnName; 
+        const newNode = buildNode('File', {
+          fileName: fnName, // the name of this function
+          fileExtension: '.mjs',
+          nodes: [
+            buildNode('Imports'),
+            buildNode('FunctionDefinition', { name: node.name, contractName }),
+          ],
+        })
+
+
+        newNode.sendTransactionNode = buildNode('SendTransaction', {
+          functionName: node.fileName,
+          contractName,
+        });
+        console.log('Node in vistor:', node);
+        node.parameters.parameters.forEach(para =>{
+        newNode.sendTransactionNode.publicInputs.push(para.name);
+})
+
+/newNode.nodes.push(newNode.sendTransactionNode);
```

Comment:
This node in the Public function AST generates the same code as we need for functions with secret states, you need to create a node separate for public functions.

Reply:
I don't think there is a need to create a new one, juste the "nodeType": "FunctionDefinitionPublic", that should be new